platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md

```diff
@@ -0,0 +1,202 @@
+# Durably Persist Self-Events Before Gossip
+
+---
+
+## Summary
+
+If a self-event is gossiped and the node is shutdown before the self-event is persisted to disk, then the node will
+branch in the hashgraph on restart by building a new self-event on the same self-event parent it had previously used.
```

Comment:
Is there a reason we cannot do a full sync with other nodes before the first event is created? That way you would know if other nodes know about an event created by you that you do not know about. If they do not then you are clear to use the last one you know about.

Reply:
Unfortunately, such a thing is not mathematically plausible (it would be super convenient if it was!).

We must be functional as long as >2/3 of nodes (by weight) are participating in the protocol, since if we are not then a cohort of attackers representing <1/3 of the network can shut things down indefinitely. Even if we exclude the existence of malicious node operators, we fundamentally lack the ability to distinguish between a node that is permanently offline and a node that is slow. This is the fundamental nature of asynchronous networks (aka the "A" in "ABFT").

In summary, we can pause for a period of time in an attempt to give nodes a chance to participate (we already have a heuristic pause for this exact purpose). But unless we are willing to pause for an arbitrarily long period of time, there exists no algorithm that can guarantee that we always learn of our self events if we gossip them without writing them to disk.